File: server/data/results/translations/plain/go/collatz_steps.go
package main

import "fmt"

func collatzSteps(n int) int {
	steps := 0
	for n != 1 {
		if n%2 == 0 {
			n = n / 2
		} else {
			n = 3*n + 1
		}
		steps = steps + 1
	}
	return steps
}

func main() {
	for _, start := range []int{6, 7, 27, 1} {
		fmt.Println(collatzSteps(start))
	}
}

File: server/data/results/translations/lite/plain/java/collatz_steps.java
public class Writing {

    public static int collatzSteps(int n) {
        int steps = 0;
        while (n != 1) {
            if (n % 2 == 0) {
                n = n / 2;
            } else {
                n = 3 * n + 1;
            }
            steps = steps + 1;
        }
        return steps;
    }

    public static void main(String[] args) {
        int[] starts = {6, 7, 27, 1};
        for (int start : starts) {
            System.out.println(collatzSteps(start));
        }
    }
}